src/containers/UploadVersionContainer.js

```diff
@@ -0,0 +1,135 @@
+import React, {useState} from 'react';
+import {withRouter} from 'react-router-dom';
+import {GET_STUDY_BY_ID} from '../state/queries';
+import {Mutation} from 'react-apollo';
+import FileUploadTarget from '../components/FileUpload/FileUploadTarget';
+import {CREATE_FILE_VERSION} from '../state/mutations';
+import {Button} from 'kf-uikit';
+
+const UploadVersionContainer = ({fileName, history, match, children}) => {
```

Comment:
Is this just a duplicate of the UploadContainer?
If it's the same, with exceptions for the content, maybe it would be easier to simply have a switch that says if it's a version uploader or new file uploader and render whatever children are passed to it within the `FileDialog--body`

Reply:
I agree, there seems like a lot of duplication here that I think can be abstracted to a general Modal component